sphinx/writers/manpage.py

```diff
@@ -319,7 +320,7 @@ def visit_reference(self, node: Element) -> None:
         self.visit_Text(node)
         self.body.append(self.defs['reference'][1])
 
-        if uri.startswith(('mailto:', 'http:', 'https:', 'ftp:')):
+        if not uri.startswith('#'):
```

Comment:
```suggestion
        if not uri.startswith('#') and is_safe_to_click:
```

Reply:
I don't think I agree. That being said, it's the existing behavior (however weird it is) and I don't have a concrete motivation to want to change it. So this is fine by me, can revisit this later.

Reply:
Actually, it's dumb from me. The `if` would be equivalent to `is_safe_to_click`, so applying my suggesting is the same as not having your PR at all.

Now, the if-body is doing some processing for mailto links and cuts the URI. After, we check if the URi is still empty or not and emit OSC if needed. I'd suggest doing something different and perhaps more flexible:

- Check whether you are using an URI scheme or not: if so, just keep the current behaviour. In particular, you cannot be an achor if you are using an URI scheme.
- For anchor references, do nothing.

Reply:
I'm not 100% sure what you mean, do you think it's fine that arbitrary URIs remain clickable? 

I don't have a strong opinion on that one. But better safe than sorry I guess, which is why this PR disables them.
I mainly care about fixing anchor-references (by not showing them at all), and making `man_show_urls` show all non-anchor URIs as text

Here are the cases, starred are the ones fixed by this current PR

refuri                  | clickable | shown with man_show_urls
------------------------|-----------|-------------------------
https:///example.com    | yes       | yes
gopher:///example.com   | no*       | yes*
#local-ref              | no*       | no

Reply:
Oh, ok, your approach is better. If this is how this is rendered, then I'm fine ! (it was just hard to imagine the things since I don't see a visual example).